api/src/Tests/Cases/CycleTest.php

```diff
@@ -15,23 +16,19 @@ protected function setUp(): void
         parent::setUp();
         $when = date('Y-m-d', strtotime('+1999 years'));
         $to = date('Y-m-d', strtotime('+2001 years'));
-        $this->cycle = $this->runSuccessJsonRequest(
-            'POST',
-            '/cycle',
-            null,
-            ['date_from' => $when, 'date_to' => $to],
-            201,
-            null,
-            '/cycle'
-        );
+        $this->cycle = testRun::testRun($this, 'POST', '/cycle')
+            ->setBody(['date_from' => $when, 'date_to' => $to])
+            ->run();
```

Comment:
Not an urgent thing, and this comment would apply in a similar way to `tearDown()` as well.

Should we interact directly with the database rather than via API call here for test setup/teardown? Although I imagine it'd be rare, this reliance on the API to create and delete objects for us to test against could result in test failures if the create or delete fails, I think.

Reply:
I debated that back and forth also. Doing actauly tests in setup/teardown seems odd but it also is  a nice clean path to do the work.  I decided on the side of having it a test just because but I am not sold either way. We can bring it up as we go forward. I know there are other tests where all these setup and such is done directly tot the database.